### PR TITLE
updates for julia 0.7

### PR DESCRIPTION
**Pull request by dpo**
Supersedes #65.

@Xx-fractale-xX The CRMR test causes NaNs. I commented it out.

### CODE REVIEW

File: test/test_cgs.jl [hunk 1]
```diff
@@ -44,7 +44,7 @@ resid = norm(r) / norm(b)
 @test(stats.solved)
 
 # Code coverage.
-(x, stats) = cgs(full(A), b)
+(x, stats) = cgs(Array(A), b)
```
Comment (abelsiqueira) [inline test/test_cgs.jl]: A is already an array here, so I think this is not required.

File: test/test_diom.jl [hunk 2]
```diff
@@ -44,7 +44,7 @@ resid = norm(r) / norm(b)
 @test(stats.solved)
 
 # Code coverage.
-(x, stats) = diom(full(A), b)
+(x, stats) = diom(Array(A), b)
```
Comment (abelsiqueira) [inline test/test_diom.jl]: Here as well

**Comment by abelsiqueira** [comment]
Thanks for updating #65. I found these two `Array(A)`, which may be from a previous version, because I think it's not needed for code coverage anymore.

**Comment by abelsiqueira** [comment]
Should we release a 0.6 version, before merging this?

**Comment by dpo** [comment]
> Should we release a 0.6 version, before merging this?

We could, but I'm not sure we have to. There's never been a release of Krylov.jl and I think we should go over each solver and check memory allocations (as in #62). I'll have a PR after this one to illustrate how for a couple of solvers. I'm also in the process of adding benchmarks (with `PkgBenchmark`).

**Comment by dpo** [comment]
Fixed.

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**